### PR TITLE
Extract local path logic and add tests.

### PR DESCRIPTION
**Pull request by zigomir**
Fixes #300 and #283.

Thanks @ajanes93 for opening a PR. Also thanks to @likun7981 for providing regex.
Only thing wrong in regex imho was assumption that windows drive letter can be of any length and thus `bitbucket:username/rep` was treated as local template. I fixed this in this PR and also added some tests.

I'm aware that running these tests on Windows won't pass because of `isAbsolute` used, but CircleCi runes on *nix so... I'm open to any suggestions.

@egoist what do you think?

### CODE REVIEW

**Comment by posva** [comment]
Maybe we should use appveyor for vue-cli to run tests on a windows machine too

**Comment by zigomir** [comment]
@posva I added appveyor which is finally passing now. I had configuration wrong and it was running against node 4 (which is fine) but there was npm v2 installed which wasn't fine.

I also found we had some race conditions in tests, reading from same directory (thanks Windows) etc. Should be all fixed in this PR now.

**Comment by posva** [comment]
@zigomir 😆 I see that you had some trouble due to the commit history. I hope it wasn't too much of a pain

File: lib/local-path.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+var path = require('path')
+
+module.exports = {
+  isLocalPath: function (templatePath) {
+    return /^[./]|(^[a-z|A-Z]:)/.test(templatePath)
```
Comment (posva) [inline lib/local-path.js]: Why is the `|` necessary?

Comment (zigomir) [inline lib/local-path.js]: Huh, might not be at all :) (regex by trial and error image here 😊)

**Comment by zigomir** [comment]
@posva hehe yeah. No worries, I think it's good that we run tests on Windows too. + I learned a bit how appveyor works.

I changed regex as you pointed out in CR. Let me know if there is anything else we should change.

Also, worth noting are weird race condition issues that were resolved here: https://github.com/vuejs/vue-cli/pull/341/commits/1f15a0184fde38f275b538edc56b8fae91a6262f

**Comment by Qquanwei** [comment]
thank your work <3
I need this feature , but it seems not publish to npm, what should I do?

**Pull request merged**